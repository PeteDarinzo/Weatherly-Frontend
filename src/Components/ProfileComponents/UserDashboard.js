import { Container, TextField, Typography } from "@mui/material";
import React, { useEffect, useState } from "react";
import { useSelector } from "react-redux";
import RangeSlider from "./RangeSlider";
import { countryList, countryListAlpha2, codeListAlpha2 } from "../../countryData";
import { makeStyles } from "@mui/styles";
import MenuItem from '@mui/material/MenuItem';
import Radio from '@mui/material/Radio';
import RadioGroup from '@mui/material/RadioGroup';
import FormControlLabel from '@mui/material/FormControlLabel';
import FormControl from '@mui/material/FormControl';
import FormLabel from '@mui/material/FormLabel';
import FormGroup from '@mui/material/FormGroup';
import Switch from '@mui/material/Switch';
import UserLocationForm from "./UserLocationForm";
import UserPreferencesForm from "./UserPreferencesForm";
import Grid from '@mui/material/Grid';


const useStyles = makeStyles({
  field: {
    marginTop: 20,
    marginBottom: 20,
    display: "block"
  }
});


const UserDashboard = ({ updateUser }) => {

  // const classes = useStyles();
  const [isLoading, setIsLoading] = useState(true);
  const userData = useSelector(store => store.user);
  const { postalCode, countryCode, minTemp, maxTemp, units, thunderstorm, drizzle, rain, snow, overcast } = userData;

  useEffect(() => {

    const data = Object.keys(userData);

    if (data.length) {
      setIsLoading(false);
    }

  }, [userData])

  if (isLoading) return (<b>Loading...</b>)

  return (
    <Container>
      <Grid container
        alignItems="center"
        justifyContent="center"
      >
        <Grid item md={6}>
          <UserPreferencesForm
            units={units}
            minTemp={minTemp}
            maxTemp={maxTemp}
            thunderstorm={thunderstorm}
            drizzle={drizzle}
            rain={rain}
            snow={snow}
            overcast={overcast}
            updateUserPreferences={updateUser} />
        </Grid>
        <Grid item md={6}>
          <UserLocationForm
            postalCode={postalCode}
            countryCode={countryCode}
            updateUserLocation={updateUser} />
        </Grid>
      </Grid>
    </Container>
  );
}

export default UserDashboard;