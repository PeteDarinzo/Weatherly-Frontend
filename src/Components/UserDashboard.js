import { Container, TextField, Typography } from "@mui/material";
import React, { useEffect, useState } from "react";
import { useSelector } from "react-redux";
import RangeSlider from "./RangeSlider";
import { countryList, countryListAlpha2, codeListAlpha2 } from "../countryData";
import { makeStyles } from "@mui/styles";
import MenuItem from '@mui/material/MenuItem';
import Radio from '@mui/material/Radio';
import RadioGroup from '@mui/material/RadioGroup';
import FormControlLabel from '@mui/material/FormControlLabel';
import FormControl from '@mui/material/FormControl';
import FormLabel from '@mui/material/FormLabel';
import FormGroup from '@mui/material/FormGroup';
import Switch from '@mui/material/Switch';
import UserLocationForm from "./UserLocationForm";
import UserPreferencesForm from "./UserPreferencesForm";


const useStyles = makeStyles({
  field: {
    marginTop: 20,
    marginBottom: 20,
    display: "block"
  }
});


const UserDashboard = ({ updateUser }) => {

  // const classes = useStyles();
  const [isLoading, setIsLoading] = useState(true);
  const userData = useSelector(store => store.user);
  const { postalCode, countryCode, minTemp, maxTemp, units, thunderstorm, drizzle, rain, snow, overcast } = userData;

  // const initialState = {
  //   postalCode,
  //   countryCode,
  //   minTemp,
  //   maxTemp,
  //   units,
  //   thunderstorm,
  //   drizzle,
  //   rain,
  //   snow,
  //   overcast
  // }

  useEffect(() => {

    if (userData) {
      //   setFormData({
      //     postalCode,
      //     countryCode,
      //     minTemp,
      //     maxTemp,
      //     units,
      //     thunderstorm,
      //     drizzle,
      //     rain,
      //     snow,
      //     overcast
      //   });
      setIsLoading(false);
    }

  }, [userData])

  // const [formData, setFormData] = useState(initialState);

  // const handleChange = (e) => {
  //   const { name, value } = e.target;
  //   setFormData(formData => ({
  //     ...formData,
  //     [name]: value
  //   }));
  //   console.log(formData);
  // }

  // function handleTempChange(tempArray) {
  //   setFormData(formData => ({
  //     ...formData,
  //     minTemp: tempArray[0],
  //     maxTemp: tempArray[1]
  //   }));
  // }

  // function handleToggle(e) {
  //   setFormData({
  //     ...formData,
  //     [e.target.name]: e.target.checked
  //   });
  //   console.log(formData);
  // }


  if (isLoading) return (<b>Loading...</b>)

  return (
    <Container>
      <UserLocationForm postalCode={postalCode} countryCode={countryCode} updateUserLocation={updateUser} />
      <UserPreferencesForm units={units} minTemp={minTemp} maxTemp={maxTemp} thunderstorm={thunderstorm} drizzle={drizzle} rain={rain} snow={snow} overcast={overcast} updateUserPreferences={updateUser} />
    </Container>
  );
}

export default UserDashboard;