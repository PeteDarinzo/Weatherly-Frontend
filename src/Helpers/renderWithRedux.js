import { render } from '@testing-library/react';
import { createStore, applyMiddleware } from "redux";
import { Provider } from 'react-redux';
import rootReducer from "../Reducers/rootReducer";
import thunk from "redux-thunk"


export default function renderWithRedux(component, { store = createStore(rootReducer, applyMiddleware(thunk)) } = {}
) {
  return {
    ...render(
      <Provider store={store}>
        {component}
      </Provider>)
  }
}