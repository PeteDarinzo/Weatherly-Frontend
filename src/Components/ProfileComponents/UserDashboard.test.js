import { MemoryRouter } from "react-router-dom";
import renderWithRedux from "../../Helpers/renderWithRedux";
import UserDashboard from "./UserDashboard";

it("renders user dashboard without crashing", () => {
  renderWithRedux(
    <MemoryRouter>
      <UserDashboard updateUser={() => { }} />
    </MemoryRouter>
  );
});


it("matches the user dashboard snapshot", () => {
  const { asFragment } = renderWithRedux(
    <MemoryRouter>
      <UserDashboard updateUser={() => { }} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
