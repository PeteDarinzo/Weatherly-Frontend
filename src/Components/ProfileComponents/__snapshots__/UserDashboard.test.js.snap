// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the user dashboard snapshot 1`] = `
<DocumentFragment>
  <b>
    Loading...
  </b>
</DocumentFragment>
`;
